moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -903,14 +903,22 @@ bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext &cont
         ROS_ERROR_STREAM_NAMED("traj_execution", "Unknown joint in trajectory: " << joint_names[i]);
         return false;
       }
+
       // TODO: check multi-DoF joints ?
-      if (fabs(current_state->getJointPositions(jm)[0] - positions[i]) > allowed_start_tolerance_)
+      double position = positions[i];
+
+      // quick check
+      if (fabs(current_state->getJointPositions(jm)[0] - position) <= allowed_start_tolerance_)
```

Comment:
I understood your https://github.com/ros-planning/moveit/pull/290#discussion_r83679715 so that this wouldn't be needed. But obviously it is. CurrentStateMonitor doesn't normalize itself.